channeldb/invoices.go

```diff
@@ -1141,11 +1141,9 @@ func deserializeHtlcs(r io.Reader) (map[CircuitKey]*InvoiceHTLC, error) {
 			return nil, err
 		}
 
-		streamBytes := make([]byte, streamLen)
-		if _, err := r.Read(streamBytes); err != nil {
-			return nil, err
-		}
-		streamReader := bytes.NewReader(streamBytes)
+		// Limit the reader so that it stops at the end of this htlc's
+		// stream.
+		htlcReader := io.LimitReader(r, streamLen)
```

Comment:
is this in any hot-path?

Reply:
not that i know of, but it will be as invoices start to have multiple htlcs or payments. might improve `listinvoices` performance slightly